realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/AuthTests.java

```diff
@@ -266,4 +270,130 @@ public void changePassword_throwWhenUserIsLoggedOut() {
         user.changePassword("new-password");
     }
 
+    // Cached instances of RealmConfiguration should not be allowed to be used if the user is no longer valid
+    @Test
+    public void cachedInstanceShouldThrowIfUserBecomeInvalid() throws InterruptedException {
+        String username = UUID.randomUUID().toString();
+        String password = "password";
+
+        SyncCredentials credentials = SyncCredentials.usernamePassword(username, password, true);
+        SyncUser user = SyncUser.login(credentials, Constants.AUTH_URL);
+        final RealmConfiguration configuration = new SyncConfiguration.Builder(user, Constants.USER_REALM).build();
+        Realm.getInstance(configuration);
```

Comment:
This instance needs to be closed, otherwise assertion will happen if you build with debug sync.